### PR TITLE
Fixed bugs with adding URL at specific commit

### PR DESCRIPTION
**Pull request by olehermanse**
- Fixed bug with adding URL at specific commit
- Added test for adding a URL at specific commit
- Fixed problems with adding/cloning the same URL and commit twice

### CODE REVIEW

**Review by craigcomstock — APPROVED** [review]

File: cfbs/core.py [hunk 1]
```diff
@@ -329,6 +330,10 @@ def _add_using_url(
             assert url.startswith(("https://", "git://", "ssh://"))
             config_path, url_commit = clone_url_repo(url)
 
+        if "@" in url:
+            assert url.split("@")[1] == url_commit
+            url = url.split("@")[0]
```
Comment (vpodzime) [inline cfbs/core.py:335]: This needs to be a bit more careful. This is the code I wrote when I was originally adding support for modules from git repos:
```
    if "@" in repo_url and (repo_url.rindex("@") > repo_url.rindex(".")):
        # commit specified in the url
        repo_url, commit = repo_url.rsplit("@", 1)
```
The reason for the complexity is that `user@some-URL` can appear in the URLs, so splitting on `@` must be done more carefully.

File: cfbs/internal_file_management.py [hunk 2]
```diff
@@ -139,6 +139,13 @@ def _get_git_repo_commit_sha(repo_path):
         return f.read().strip()
 
 
+def _clone_and_checkout(url, path, commit):
+    # NOTE: If any of these shell (git) commands fail, we will exit
+    if not os.path.exists(path + "/.git"):
```
Comment (vpodzime) [inline cfbs/internal_file_management.py:144]: Please use `os.path.join()`.

Comment (olehermanse) [reply cfbs/internal_file_management.py]: Fixed, thanks!

File: cfbs/core.py (hunk 1 above)
Comment (olehermanse) [reply cfbs/core.py]: Thank you, fixed. See last 2 commits.

**Pull request merged**